README.md

```diff
@@ -145,7 +184,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+Solution is in "main.cpp" in file. Compile and run "main.cpp" to run unit tests.
```

Comment:
# 5

It passes its tests, but there isn't a state machine in there. It's all linear code with "if"s.
I was looking for something like

~~~ cpp
#include <iostream>
using namespace std;
#include <map>

typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

const int END = 99;

state_table states = {
  { 0, {{ "the", 1 }}},
        { 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
        { 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
        { 3, {{ "ate", 4},  {"ran",    4}}},
        { 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
        { 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
        transitions t = states[currentState];
        if (t.count(nextWord))
                return t[nextWord];
        else
                return -1;
}
~~